### PR TITLE
Use CRLF for C++ files.

### PR DESCRIPTION
**Pull request by JunielKatarn**
Editing/convenience change.

There is an inconsistency between EditorConfig settings and Git attributes for C++ files.
This change ensures all local C++ files use the CRLF line ending.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/Microsoft/react-native-windows/pull/2358)

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -2,9 +2,12 @@
 # Auto detect text files and perform LF normalization
 
 *       text=auto
-*.cpp   -text
-*.h     -text
 
+# C++ files should use CRLF.
```
Comment (akrantz) [inline .gitattributes]: Why do you need to force C++ files to be different than any other text file?  I have to go back and read up on the .gitattribute values as I cannot remember off of the top of my head, but basically Windows users probably have the config to do autocrlf, and I think `text=auto` honors that.

Comment (akrantz) [reply .gitattributes]: Well I expanded the diff and I think the answer is in the comment below.

Comment (JunielKatarn) [reply .gitattributes]: git@github.com:JunielKatarn/react-native-windows.git

Comment (JunielKatarn) [reply .gitattributes]: Pretty much.
We shouldn't assume the developer has "canonical" settings just because they are using Windows.

**Comment by ahimberg** [comment]
Is this going to fix the entire-file whitespace diffs we're seeing on lots of pull requests since the last change?

**Pull request closed without merging**